test/stub-node-only.js

```diff
@@ -1133,48 +1173,63 @@ describe("tests that only work against stub server", function () {
               return "0x" + ethereumjsAbi.rawEncode(["uint256[]"], [[1, 100, 100000]]).toString("hex");
             }
           });
-          rpc.callContractFunction(payload, callback, wrapper, expectedObject);
+          rpc.callContractFunction(payload, function (err, result) {
+            assert.strictEqual(result[0], "0x0000000000000000000000000000000000000000000000000000000000000001");
+            assert.strictEqual(result[1], "0x0000000000000000000000000000000000000000000000000000000000000064");
+            assert.strictEqual(result[2], "0x00000000000000000000000000000000000000000000000000000000000186a0");
+            server.assertExpectations();
+            done();
+          });
         });
 
         it("transact pool not accepting", function (done) {
           function onSent() { }
           function onSuccess() { assert.isFalse(true, "onSuccess should not have been called"); }
-          function onFailure(error) {
-            assert.strictEqual(error.error, 413);
-            assert.strictEqual(error.hash, "0xbadf00dbadf00dbadf00dbadf00dbadf00dbadf00dbadf00dbadf00dbadf" + "0006");
+          function onFailed(err) {
+            assert.strictEqual(err.error, 413);
+            assert.strictEqual(err.hash, "0xbadf00dbadf00dbadf00dbadf00dbadf00dbadf00dbadf00dbadf00dbadf" + "0006");
             done();
           }
           server.addResponder(function (jso) { if (jso.method === "eth_call") return "0x12"; });
           server.addResponder(function (jso) { if (jso.method === "eth_getTransactionByHash") return null; });
-          rpc.transact(createReasonableTransactPayload(), null, null, onSent, onSuccess, onFailure);
+          rpc.transact(createReasonableTransactPayload(), null, null, onSent, onSuccess, onFailed);
         });
 
         it("transact", function (done) {
-          function onSent(result) { }
+          function onSent() { }
           function onSuccess(result) {
             assert.strictEqual(result.callReturn, "18");
             done();
           }
-          function onFailure(error) { assert.isFalse(true, "onFailure should not have been called."); }
+          function onFailed(error) {
+            assert.isFalse(true, "onFailed should not have been called." + error);
+          }
           server.addResponder(function (jso) {
-            if (jso.method === "eth_call") return "0x12";
+            switch (jso.method) {
+              case "eth_call":
+                return "0x12";
+              case "eth_getTransactionReceipt":
+                return { status: "0x1" };
+            }
```

Comment:
Seems like this switch statement should maybe have a default specified.

Reply:
In this case, a default isn't needed, because it "falls thru" to the other responders already registered with the stub server.